### ISSUE TITLE
QAT export of custom model fails

### ISSUE THREAD

**Issue by NelmerXp** [google-ai-edge/ai-edge-torch#364]
### Description of the bug:

Background: I am trying to implement quantization aware training in deep reinforcement learning with [stable baselines 3](https://github.com/DLR-RM/stable-baselines3/tree/master).

I am preparing my custom model for QAT following implementation according to the [newest tutorial](https://pytorch.org/tutorials/prototype/pt2e_quant_qat.html). 

```  
    example_inputs = (th.randn(1, observation_size),)
    exported_model = th.export.export_for_training(float_model, example_inputs).module()

    quantizer = XNNPACKQuantizer()                                  
    quantizer.set_global(get_symmetric_quantization_config(is_qat=True))

    prepared_model = prepare_qat_pt2e(exported_model, quantizer)
    prepared_model_train_eval = allow_exported_model_train_eval(prepared_model)
```

The model is training just fine and it is also usable with torch after training. 

My export script looks like this: 
```
model = DQN.load(log_path / agent_name, env)
observation_size = model.observation_space.shape[0]
example_inputs = (torch.randn(1, observation_size),)
model.policy.q_net.q_net = convert_pt2e(model.policy.q_net.q_net, fold_quantize=False)
quantized_export = torch.export.export(model.policy, example_inputs)

torch.export.save(quantized_export, str(model_path + ".pth"))
loaded_qant_export = torch.export.load(str(model_path + ".pth"))
loaded_qant_model = loaded_qant_export.module()

quantizer = XNNPACKQuantizer()                                  
quantizer.set_global(get_symmetric_quantization_config(is_qat=True))
tflite_model = ai_edge_torch.convert(
    loaded_qant_model, example_inputs, quant_config=QuantConfig(pt2e_quantizer=quantizer),
)
```

### Actual vs expected behavior:

Instead of converting I firstly get the error regarding the "is_dynamic" discussed in #355. After fixing that issue I now run into a very lengthy error message. The gist of it being: 
```
 tflite_model = ai_edge_torch.convert(
  File "/usr/local/lib/python3.10/dist-packages/ai_edge_torch/convert/converter.py", line 195, in convert
    return Converter().convert(
  File "/usr/local/lib/python3.10/dist-packages/ai_edge_torch/convert/converter.py", line 134, in convert
    return conversion.convert_signatures(
  File "/usr/local/lib/python3.10/dist-packages/ai_edge_torch/convert/conversion.py", line 120, in convert_signatures
    tflite_model = cutils.convert_stablehlo_to_tflite(
  File "/usr/local/lib/python3.10/dist-packages/ai_edge_torch/convert/conversion_utils.py", line 412, in convert_stablehlo_to_tflite
    tflite_model = converter.convert()
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1231, in wrapper
    return self._convert_and_export_metrics(convert_func, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1183, in _convert_and_export_metrics
    result = convert_func(self, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1562, in convert
    return self._convert_from_saved_model(graph_def)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1423, in _convert_from_saved_model
    result = _convert_saved_model(**converter_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert_phase.py", line 212, in wrapper
    raise converter_error from None  # Re-throws the exception.
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert_phase.py", line 205, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert.py", line 1031, in convert_saved_model
    data = convert(
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert.py", line 376, in convert
    raise converter_error
tensorflow.lite.python.convert_phase.ConverterError: Could not translate MLIR to FlatBuffer.-:0:0: error: unknown vhlo type code: 33
-:0:0: note: in bytecode version 6 produced by: StableHLO_v1.7.1
-:0:0: error: unknown vhlo type code: 33
-:0:0: note: in bytecode version 6 produced by: StableHLO_v1.7.1
<unknown>:0: error: loc(callsite(callsite(fused["XlaCallModule:", "XlaCallModule@__inference_inner_35"] at fused["StatefulPartitionedCall:", "StatefulPartitionedCall@__inference_signature_wrapper_53"]) at fused["StatefulPartitionedCall:", "StatefulPartitionedCall"])): 'tf.XlaCallModule' op is neither a custom op nor a flex op
<unknown>:0: note: loc(fused["StatefulPartitionedCall:", "StatefulPartitionedCall"]): called from
<unknown>:0: note: loc(callsite(callsite(fused["XlaCallModule:", "XlaCallModule@__inference_inner_35"] at fused["StatefulPartitionedCall:", "StatefulPartitionedCall@__inference_signature_wrapper_53"]) at fused["StatefulPartitionedCall:", "StatefulPartitionedCall"])): Error code: ERROR_NEEDS_CUSTOM_OPS
<unknown>:0: error: failed while converting: 'main': 
Some ops in the model are custom ops, See instructions to implement custom ops: https://www.tensorflow.org/lite/guide/ops_custom 
Custom ops: XlaCallModule
```

XlaCallModule seems to be some kind of delegation. I do not know where I introduce it and how I can get around the issue. 

### Any other information you'd like to share?

The custom net i am using has the following structure:
```
activation_fn = nn.ReLU

if len(net_arch) > 0:
        modules = [nn.Linear(input_dim, net_arch[0], bias=with_bias), activation_fn()]
    else:
        modules = []

    for idx in range(len(net_arch) - 1):
        modules.append(nn.Linear(net_arch[idx], net_arch[idx + 1], bias=with_bias))
        modules.append(activation_fn())

    if output_dim > 0:
        last_layer_dim = net_arch[-1] if len(net_arch) > 0 else input_dim
        modules.append(nn.Linear(last_layer_dim, output_dim, bias=with_bias))

q_net = nn.Sequential(*modules)
```
So far I have only worked on one environment:
torch, torch-xla==2.5.1
ai-edge-torch==0.2.1
tensorflow==2.17.0

**Comment by gurkirt** [comment]
You can try suggestion for pkgoogle from https://github.com/google-ai-edge/ai-edge-torch/issues/358

**Comment by NelmerXp** [comment]
Update: The Issue was caused by my device usage. I trained the model on an external gpu enabled workstation. I tried to apply the conversion on my personal device. Even when explicitly deploying the model on my devices cpu, my device usage led to the issue posted above. 
However when using the code posted above on the same device i trained the model the conversion works without any issue. Thus this issue is closed.
